onnxscript/tests/models/pnp.py

```diff
@@ -0,0 +1,262 @@
+            pred_roi_tensor = onnxscript_tensor.Tensor(pred_roi[0])
+            # pred_roi_tensor = onnxscript_tensor.Tensor(input.value + np.ones(input.shape, dtype=input.dtype))
+            if isinstance(input, np.ndarray):
+                return pred_roi_tensor.value
+            else:
+                return pred_roi_tensor
+
+        return op(input)
+
+op = Opset18Ext()
+
+@script()
+def extend_roi_indices(roi_indices: INT64["roi_D", "roi_H", "roi_W", 3], seg_C: INT64) -> INT64[1, "seg_C", "roi_D", "roi_H", "roi_W", 5]:
+    """
+    This function is used to extend the roi indices to the segment dimension.
+    """
+    zero = op.Constant(value_int=0)
+    one = op.Constant(value_int=1)
+    one_ = op.Constant(value_ints=[1])
+    roi_D, roi_H, roi_W, _ = op.Split(op.Shape(roi_indices), num_outputs=4)
+    shape_1_seg_C_roi_D_H_W_1 = op.Concat(one_, seg_C, roi_D, roi_H, roi_W, one_, axis=0)
+    zeros_1_seg_C_D_H_W_1 = op.Cast(op.ConstantOfShape(shape_1_seg_C_roi_D_H_W_1), to=INT64.dtype) # seg_C, roi_D, roi_H, roi_W, 1
+
+    seg_indices = op.Range(zero, seg_C, one)
+    seg_indices_unsqueezed = op.Unsqueeze(seg_indices, op.Constant(value_ints=[0, 2, 3, 4, 5])) # 1, seg_C, 1, 1, 1, 1
+    seg_indices_1_C_D_H_W_1 = seg_indices_unsqueezed + zeros_1_seg_C_D_H_W_1 # 1, seg_C, roi_D, roi_H, roi_W, 1
+    roi_indices_unsqueezed = op.Unsqueeze(roi_indices, op.Constant(value_ints=[0, 1])) # 1, 1, roi_D, roi_H, roi_W, 3
+    roi_indices_1_C_D_H_W_3 = roi_indices_unsqueezed + zeros_1_seg_C_D_H_W_1 # 1, seg_C, roi_D, roi_H, roi_W, 3 
+
+    extended_roi_indices = op.Concat(zeros_1_seg_C_D_H_W_1, seg_indices_1_C_D_H_W_1, roi_indices_1_C_D_H_W_3, axis=-1)
+    return extended_roi_indices
+
+@script()
+def roi_indices_3d(start: INT64[3], stop: INT64[3], step: INT64[3]) -> (INT64["roi_D", "roi_H", "roi_W"], INT64[3]):
+    """
+    This function is used to generate the indices for a 3d roi window.
+    """
+    start_d, start_h, start_w = op.Split(start, num_outputs=3)
+    stop_d, stop_h, stop_w = op.Split(stop, num_outputs=3)
+    step_d, step_h, step_w = op.Split(step, num_outputs=3)
+    
+    grid_d_0 = op.Range(start_d, stop_d, step_d)
+    grid_h_0 = op.Range(start_h, stop_h, step_h)
+    grid_w_0 = op.Range(start_w, stop_w, step_w)
+
+    D = op.Shape(grid_d_0)
+    H = op.Shape(grid_h_0)
+    W = op.Shape(grid_w_0)
+
+    roi_shape = op.Concat(D, H, W, axis=0)
+    zeros_D_H_W = op.CastLike(op.ConstantOfShape(roi_shape), start)
+
+    zeros_H_W_D = op.Transpose(zeros_D_H_W, perm=[1, 2, 0])
+    grid_d_1 = zeros_H_W_D + grid_d_0
+    grid_d = op.Transpose(grid_d_1, perm=[2, 0, 1])
+
+    zeros_D_W_H = op.Transpose(zeros_D_H_W, perm=[0, 2, 1])
+    grid_h_1 = zeros_D_W_H + grid_h_0
+    grid_h = op.Transpose(grid_h_1, perm=[0, 2, 1])
+
+    grid_w = grid_w_0 + zeros_D_H_W
+
+    indices_seq = op.SequenceConstruct(grid_d, grid_h, grid_w)
+    indices = op.ConcatFromSequence(indices_seq, axis=-1, new_axis=1)  # [D, H, W, 3]

```

Comment:
I guess you need this because Concat doesn't support the new_axis attributes? I wonder if we should extend Concat to support it.

We could explicitly unsqueeze grid_d etc. to add the extra dimension at end and then use Concat. It is a bit more cumbersome, but onnxruntime doesn't handle sequence op that well (it may create extra copies of the tensor). Anyway, may not be worth bothering about for now.